### ISSUE TITLE
export more of trigger attributes into triggers.txt file

### ISSUE THREAD

**Issue by rusefillc** [rusefi/rusefi#2077]
At the moment most of trigger parameters are defined in C code and exported into triggers.txt for TriggerImage java code to render, but some stuff is hard coded in java

1) human-readable trigger name
2) isFirstCrankBased
3) isSecondCamBased

all data about trigger should be defined in one place and at the moment that one place is C trigger definition structure

We should remove all hard-coded stuff from java and hard-code anything missing in C

**Comment by rusefillc** [comment]
```
; todo: code generator to hard-code all triggers not requiring VVT?	    
	    field = "CAM/VVT mode shaft #1",				vvtMode, {trigger_type != TT_TT_MAZDA_MIATA_NA}
```

that's another reason to flip trigger workflow

2021 idea: instead of C generating triggers.txt we shall have triggers.yaml generating C and java and .ini

**Comment by rusefillc** [comment]
@chuckwagoncomputing how can I trick you into this issue?

**Comment by chuckwagoncomputing** [comment]
> instead of C generating triggers.txt we shall have triggers.yaml generating C and java and .ini

Which .ini?

**Comment by rusefillc** [comment]
.input
https://github.com/rusefi/rusefi/blob/master/firmware/tunerstudio/rusefi.input#L1808
and
https://github.com/rusefi/rusefi/blob/master/firmware/tunerstudio/rusefi.input#L1820

.txt
https://github.com/rusefi/rusefi/blob/master/firmware/integration/rusefi_config.txt#L653

**Comment by chuckwagoncomputing** [comment]
So the trigger info should be taken out of those files, and the yaml should be passed to ConfigDefinition.
Do I understand?

**Comment by rusefillc** [comment]
Also most of https://github.com/rusefi/rusefi/tree/master/firmware/controllers/trigger/decoders should be replaced by something generated by new java code based on triggers.yaml

And https://github.com/rusefi/rusefi/blob/master/java_console/ui/src/main/java/com/rusefi/TriggerImage.java#L50

**Comment by rusefillc** [comment]
by the way firing order tool is somewhat similar just much simpler

**Comment by rusefillc** [comment]
Depends on #2607

**Comment by rusefillc** [comment]
Wow a sad discovery #4032

**Comment by chuckwagoncomputing** [comment]
What is the current state of this?

**Comment by rusefillc** [comment]
a lot of progress was done!

at least nice names are still in java and should be moved into firmware as EFI_UNIT_TEST field? Actually human-readable name could be the last thing to move out of java

```
    private static String getTriggerName(TriggerWheelInfo triggerName) {
        switch (findByOrdinal(triggerName.getId())) {
            case TT_FORD_ASPIRE:
                return "Ford Aspire";
            case TT_VVT_BOSCH_QUICK_START:
                return "Bosch Quick Start";
            case TT_MAZDA_MIATA_NA:
                return "Miata NA";
            case TT_MAZDA_MIATA_NB1:
                return "Miata NB";
            case TT_SUBARU_SVX:
                return "Subaru SVX";
            case TT_HONDA_K_CRANK_12_1:
                return "Honda K 1/12";
            case TT_SUBARU_7_6:
                return "Subaru 7/6";
            case TT_GM_24x:
```

**Comment by chuckwagoncomputing** [comment]
I'm trying to decide what to do with initializeSkippedToothTrigger.
We can either:
- Get rid of initializeSkippedToothTrigger and define all triggers as list of events - Already done for existing triggers, but more work to create new triggers
- Create an implementation of initializeSkippedToothTrigger in Java for TriggerImage
@rusefillc Do you have any input?

**Comment by rusefillc** [comment]
@chuckwagoncomputing I am out of context

Why does anything need to happen to initializeSkippedToothTrigger helped method? how is initializeSkippedToothTrigger related to TriggerImage?

**Comment by chuckwagoncomputing** [comment]
This is related to generating C trigger initialization code from a YAML trigger definition file.

initializeSkippedToothTrigger allows triggers to be defined by specifying a tooth count and skipped tooth count, instead of providing a list of tooth angles.

TriggerImage would need an implementation of initializeSkippedToothTrigger to be able to generate images for triggers defined in the YAML using this method.

**Comment by rusefillc** [comment]
I think that this ticket should be aborted.

trigger generation/trigger manipulation is well below-average broken. Triggers are rarely added these days, when they are added the labor is less about coding.

"we can make this YAML but this would create more work" sounds like YAML is a bad idea, sounds like the change is not worth it. Sounds like we need pretty complex language/tool for comfortable like, which is just not worth being re-created comparing from current C++ which just works
